app/src/main/kotlin/com/wire/android/ui/calling/SharedCallingViewModel.kt

```diff
@@ -135,18 +137,20 @@ class SharedCallingViewModel @Inject constructor(
 
     private suspend fun observeOnMute() {
         snapshotFlow { callState.isMuted }.collectLatest {
-            if (it) {
-                muteCall(conversationId)
-            } else {
-                unMuteCall(conversationId)
+            it?.let {
+                if (it) {
+                    muteCall(conversationId)
+                } else {
+                    unMuteCall(conversationId)
+                }
             }
         }
     }
 
     private suspend fun observeCallState() {
         allCalls().collect { calls ->
             calls.find { call ->
-                call.conversationId == conversationId && call.status == CallStatus.ESTABLISHED
+                call.conversationId == conversationId && call.status != CallStatus.CLOSED
```

Comment:
```suggestion
                call.conversationId == conversationId && call.status != CallStatus.CLOSED && call.status != CallStatus.MISSED
```

Reply:
We also need to check if it is not a MISSED call

Reply:
good catch thanks